### PR TITLE
Add @Inherited

### PR DESCRIPTION
**Pull request by nipafx**
Proposed commit message:

```
Document and improve reusability of various extensions (#456 / #634)

Expand contribution guide with explanation how to make extensions
reusable, namely:

* allow as meta-annotation and use our and Jupiter's tools to find
  annotations
* allow inheritance by applying `@Inherited`

This change also applies `@Inherited` where it wasn't already.

Closes: #456
PR: #634
```

---
**PR checklist**

The following checklist shall help the PR's author, the reviewers and maintainers to ensure the quality of this project.
It is based on our contributors guidelines, especially the ["writing code" section](https://github.com/junit-pioneer/junit-pioneer/blob/main/CONTRIBUTING.md#writing-code).
It shall help to check for completion of the listed points.
If a point does not apply to the given PR's changes, the corresponding entry can be simply marked as done. 

Documentation (general)
* [x] There is documentation (Javadoc and site documentation; added or updated)
* [x] There is implementation information to describe _why_ a non-obvious source code / solution got implemented
* [x] Site documentation has its own `.adoc` file in the `docs` folder, e.g. `docs/report-entries.adoc`
* [x] Site documentation in `.adoc` file references demo in `src/demo/java` instead of containing code blocks as text
* [x] Only one sentence per line (especially in `.adoc` files)
* [x] Javadoc uses formal style, while sites documentation may use informal style

Documentation (new extension)
* [x] The `docs/docs-nav.yml` navigation has an entry for the new extension
* [x] The `package-info.java` contains information about the new extension

Code
* [x] Code adheres to code style, naming conventions etc.
* [x] Successful tests cover all changes
* [x] There are checks which validate correct / false usage / configuration of a functionality and there are tests to verify those checks
* [x] Tests use [AssertJ](https://assertj.github.io/doc/) or our own [PioneerAssert](https://github.com/junit-pioneer/junit-pioneer/blob/main/CONTRIBUTING.md#assertions) (which are based on AssertJ)

Contributing
* [x] A prepared commit message exists
* [x] The list of contributions inside `README.md` mentions the new contribution (real name optional)

### CODE REVIEW

**Review by Bukama — APPROVED** [review]
LGTM 👍 
closes #456 (creating link :D )

**Review by Michael1993 — APPROVED** [review]

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -183,6 +183,19 @@ How to write the code itself.
...
+We strive to make our extensions reusable and extensible.
+
+A key ingredient in that is making sure that annotations work as meta-annotations (i.e. users can apply _our_ annotations to _their_ annotations and our extensions still work).
+To achieve this, apply `@Target({ ElementType.ANNOTATION_TYPE })` to annotations and prefer `org.junitpioneer.internal.PioneerAnnotationUtils` and `org.junit.platform.commons.support.AnnotationSupport` when searching for annotations.
+
+Another aspect is that annotations that apply to classes (i.e. those marked with `@Target({ ElementType.TYPE })`) should be inherited by subclasses.
+For that, also add the annotation `@Inherited`.
+
+**NOTE**:
+`ElementType.TYPE` includes annotations, so there's no need to apply it _and_ `ElementType.ANNOTATION_TYPE`.
```
Comment (beatngu13) [inline CONTRIBUTING.md:197]: Nice, I learned something! 😊

**Review by beatngu13 — APPROVED** [review]

**Comment by beatngu13** [comment]
Experimental builds on JDK 18 [complain](https://github.com/junit-pioneer/junit-pioneer/runs/6455499109?check_suite_focus=true#step:8:68):

> Task :checkstyleTest
>Error: eckstyle] [ERROR] /home/runner/work/junit-pioneer/junit-pioneer/src/test/java/org/junitpioneer/testkit/PioneerTestKit.java:70:12: Unable to get class information for @throws tag 'TestAbortedException'. [JavadocMethod]
>Error: eckstyle] [ERROR] /home/runner/work/junit-pioneer/junit-pioneer/src/test/java/org/junitpioneer/testkit/PioneerTestKit.java:70:12: Unused @throws tag for 'TestAbortedException'. [JavadocMethod]

As of yesterday, this is also the case on [`main`](https://github.com/junit-pioneer/junit-pioneer/runs/6447896576?check_suite_focus=true#step:8:68).

**Comment by nipafx** [comment]
Seems to be unrelated to this change, so I'll merge and fix that later. If not before, then on Thursday's stream.

**Pull request merged**